### ISSUE TITLE
semicolon rule does not detect semicolons in nested properties in interfaces

### ISSUE THREAD

**Issue by pavelkucera** [palantir/tslint#3955]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.9.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export interface Test {
    child: {
        first: string;
        second: string;
    }
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "semicolon": [true, "never"]
  }
}

```

#### Actual behavior
No error is reported.

#### Expected behavior
Both semicolons should be reported as unnecessary.

If you can verify that this is a bug, I will do my best to send a PR with a fix.

**Comment by giladgray** [comment]
@pavelkucera sure, we can call this a bug. go ahead with that PR.

(i use prettier with tslint in my projects, which supersedes this rule and enforces all the semicolons.)

**Comment by pavelkucera** [comment]
@giladgray I looked into the issue and currently I do not see a way of fixing this, without changing interfaces declared by TypeScript. All the `Semicolon*Walker.checkInterface` work with typescript's `InterfaceDeclaration.members` which do not provide the type of a property, only its "PropertySignature" kind, and thus an object literal can not be recognised as far as I understand it.

I will report this at TypeScript. Any advice how to work around this?

**Comment by giladgray** [comment]
@pavelkucera if typescript does not provide this information then there's nothing to be done.
